api/database.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * Promise database implementation, with stricter typing.
+ * By Mia
+ * @author mia-pi-git
+ */
+import * as mysql from 'mysql';
+import {Config} from './config-loader';
+
+export type SQLInput = string | number | null;
+export type ResultRow = {[k: string]: SQLInput};
+
+export const databases: PSDatabase[] = [];
+
+export class PSDatabase {
+	pool: mysql.Pool;
+	prefix: string;
+	constructor(config: {[k: string]: any} = Config.mysql) {
+		this.pool = mysql.createPool(config);
+		this.prefix = config.prefix === true ? "ntbb_" : config.prefix;
+		if (!databases.includes(this)) databases.push(this);
+	}
+	query<T = ResultRow>(queryString: string, args: SQLInput[]) {
+		return new Promise<T[]>((resolve, reject) => {
+			this.pool.query(queryString, args, (e, results) => {
+				// conn.active = false;
+				if (e) {
+					return reject(
+						e.sqlMessage ? new Error(`${e.sqlMessage} ('${e.sql}')`) : new Error(e.message)
+					);
+				}
+				if (Array.isArray(results)) {
+					for (const chunk of results) {
+						for (const k in chunk) {
+							if (Buffer.isBuffer(chunk[k])) chunk[k] = chunk[k].toString();
+						}
+					}
+				}
+				return resolve(results);
+			});
+		});
+	}
+	async get<T = ResultRow>(queryString: string, args: SQLInput[]): Promise<T | null> {
+		// if (!queryString.includes('LIMIT')) queryString += ` LIMIT 1`;
+		// limit it yourself, consumers
+		const rows = await this.query(queryString, args);
+		if (Array.isArray(rows)) return rows[0] as unknown as T;
+		return rows || null;
```

Comment:
This will convert a result of `0` or `''` to null; is this desired? What are you trying to convert to null here? If it's jut `undefined`, `??` would be better.